### PR TITLE
Add cross compiling support

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@jakirkham, @kkraus14, can you check that this doesn't break rerendering cuda enabled feedstocks?

**Comment by jakirkham** [comment]
Where do these environment variables get set?

**Comment by isuruf** [comment]
They are set by conda-smithy. https://github.com/conda-forge/conda-smithy/pull/1360/files#diff-3a2c4a048580db27ad9db93e2780b511R566-R570

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -110,23 +110,23 @@ channel_sources:
 channel_targets:
   - conda-forge main
 
-docker_image:                                   # [linux]
-  - condaforge/linux-anvil-comp7                # [linux64]
-  - condaforge/linux-anvil-cuda:9.2             # [linux64]
-  - condaforge/linux-anvil-cuda:10.0            # [linux64]
-  - condaforge/linux-anvil-cuda:10.1            # [linux64]
-  - condaforge/linux-anvil-cuda:10.2            # [linux64]
+docker_image:                           # [(os.environ.get("CONFIG_VERSION", "1") == "1" and linux) or os.environ.get("BUILD_PLATFORM", "").startswith("linux")]
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:113]: I added a default value of `CONFIG_VERSION=1` for older conda-smithy.

**Comment by scopatz** [comment]
The selectors would be easier to read if the build platform check were put first.  Also, at some point when you have multiple lines with the same complex selectors, it might just be easier to use jinja if-blocks than selectors

**Comment by isuruf** [comment]
> Also, at some point when you have multiple lines with the same complex selectors, it might just be easier to use jinja if-blocks than selectors

I don't think `conda_build_config.yaml` allows jinja.

**Comment by jjhelmus** [comment]
> I don't think `conda_build_config.yaml` allows jinja.

You are correct, jinja is [not allowed](https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#using-variants-with-the-conda-build-api) in `conda_build_config.yaml` files.  To quote from the docs:

> Jinja is not allowed in conda_build_config.yaml, though. It is the source of information to feed into other Jinja templates, and the buck has to stop somewhere.

**Comment by isuruf** [comment]
Ping on this

**Comment by jakirkham** [comment]
Sorry Isuru! Have been a bit swamped of late. Will try to allocate some time to look at this tomorrow 🙂

**Comment by jakirkham** [comment]
Sorry for the delay. Have been caught up with CUDA 11 work.

Seems to be working ok AFAICT (assuming it is already deployed).

xref: https://github.com/conda-forge/nccl-feedstock/pull/25
xref: https://github.com/conda-forge/ucx-split-feedstock/pull/74

**Comment by isuruf** [comment]
Thanks @jakirkham

**Pull request merged**